### PR TITLE
Feature/api calls krd

### LINKED ISSUE #5: Api calls

### PR DESCRIPTION
**Pull request by krdurrett**
## What Features are Being Added?
Added api/network requests in an apiCalls file.
Added a utils.js file to normalize the data coming back from our network requests.

## What is Being Refactored?
Refactored App.js to include componentDidMount to call our initial network get request.

## Where should the reviewer start?
apiCalls.js

## How should this be manually tested?
Once we add CORS to api repo/deployment, you'll be able to see our fetch call working on our home page.

## Any Other Notes/Questions/Problems?
We have to add CORS to api and redeploy

### CODE REVIEW

File: src/apiCalls.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export const apiCalls = {
```
Comment (cagallo) [inline src/apiCalls.js:1]: Great job with these API calls! Smart to get them all done early.

File: src/components/App.js [hunk 2]
```diff
@@ -87,6 +89,14 @@ class App extends Component {
     }
   }
 
+  componentDidMount = () => {
+    apiCalls.getAllBooks()
+      .then(booksData => {
+        const cleanedBooksData = booksData.map(book => cleanBookData(book));
+        this.setState({ books: cleanBookData})
+      }) 
+  }
```
Comment (cagallo) [inline src/components/App.js:98]: Need to add our catch but looks good! Can test more once the CORS issue fixed.

File: src/utils.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+export const cleanBookData = (book) => {
+  return  {
+    id: book.id,
+    isbn: book.isbn,
+    title: book.title,
```
Comment (ectweitmann) [inline src/utils.js:5]: It'll be a fun challenge to figure out the logic of how to format this title correctly!

File: src/utils.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    description: book.description,
+    amazon_link: book['amazon_link'],
```
Comment (cagallo) [inline src/utils.js:7]: Smart to normalize our data in this file!

**Review by cagallo — APPROVED** [review]

File: src/apiCalls.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  getAllBooks() {
```
Comment (Fordo29) [inline src/apiCalls.js:2]: Nice to see how to create apiCalls into an object to export

File: src/utils.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+export const cleanBookData = (book) => {
```
Comment (Fordo29) [inline src/utils.js:1]: Nice cleaning of data

File: src/apiCalls.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+      body: JSON.stringify(favoriteBook),
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    }).then(response => response.json())
+  },
+  deleteFromFavorites(id) {
+    return fetch(`https://book-club-api-2110.herokuapp.com/api/v1/favorites/${id}`, { method: 'DELETE' })
+      .then(response => response.json())
+  }
```
Comment (ectweitmann) [inline src/apiCalls.js:26]: Nice I love making this an object!

**Review by ectweitmann — APPROVED** [review]
Looks great, let's keep it up!

**Pull request merged**